app/models/miq_report/search.rb

```diff
@@ -104,7 +104,9 @@ def paged_view_search(options = {})
 
     if options[:parent]
       targets = get_parent_targets(options)
-      if targets.empty?
+      # filter out archived VMs
+      targets = targets.select { |target| !Vm.find_by(:id => target).try(:archived?) }
```

Comment:
I think this change is a problem for a couple of reasons. First, paged_view_search is generically used for searching on any kind of object. Not just VMs. So, there's no guarantee that the ids in targets will be the ids of Vm objects. Second, looking up each Vm by id one at a time will create an n+1 query performance issue.

I see that you are filtering out the archived Vms here https://github.com/ManageIQ/manageiq/pull/5410/files#diff-6e332326f5467e2bd25aa3441a9cba33R36, anyway. So, maybe it's ok to just remove this change. Otherwise, additional filtering options would need to be passed down through the callers.